### ISSUE TITLE
Datatable ui-state-highlight class automatically added to rows

### ISSUE THREAD

**Issue by viedje** [primefaces/primeng#3414]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

**Current behavior**
I have implemented the current workaround in the issue #1681 for hiding checkbox for some rows depending on some condition when multiple selection mode is selected for the datatable.
When adding the CSS class (for hide/show) with the rowStyleClass property, the class ui-state-highlight is automatically added to the row and stays even with the class (for hide/show) removed. 
That makes the row looks as if it is selected.

**Expected behavior**
ui-state-highlight class should not be added when adding or removing a class to hide or show the checkbox for the row

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 4.0.2

* **PrimeNG version:** 4.0.2

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by mlakhara** [comment]
The same thing is happening with me when I use the rowStyleClass property. If the function returns anything (including null and undefined), it starts behaving this way. On investigation I found that all the classes ui-datatable-even, ui-datatable-odd, ui-datatable-highlight were added to the row. (This are present in ngClass attribute on tr element in the template of datatable tBody). My guess was that using [class] along with ngClass is causing an issue. I tried changing the datatable.js code so that all the classes are being returned from the getRowStyleClass method of datatable and I removed the [ngClass] directive from the template. This behaved in the expected way.

**Comment by mlakhara** [comment]
Is this the same as #1827 ?

**Comment by cagataycivici** [comment]
Fixed via 1827 yes.

**Comment by viedje** [comment]
Effectively it has been fixed in the 4.1.3 version.
Thanks
